packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -83,12 +83,15 @@ function classOrObjectMethod(path: NodePath, callId: Object) {
 
   node.async = false;
 
-  const container = t.functionExpression(null, [], t.blockStatement(body.body), true);
+  const container = t.functionExpression(node.key, [], t.blockStatement(body.body), true);
```

Comment:
This will fail if the method name isn't a valid function name, or if the method is a computed property, e.g.

```
class Foo {
  async finally(){}
}
```
or
```
class Foo {
  async ["met" + "hod"(){}
}
```
or
```
class Foo {
  async "method"(){}
}
```